addons/account_edi/models/account_move.py

```diff
@@ -226,35 +226,78 @@ def _serialize_python_dictionary(vals):
         def default_grouping_key_generator(tax_values):
             return {'tax': tax_values['tax_id']}
 
+        def compute_invoice_lines_tax_values_dict_from_sql_query(invoice_lines):
+            invoice_lines_tax_values_dict = defaultdict(list)
+            domain = [('move_id', '=', self.id)]
+            tax_details_query, tax_details_params = invoice_lines._get_query_tax_details_from_domain(domain)
+            self._cr.execute(tax_details_query, tax_details_params)
+            for row in self._cr.dictfetchall():
+                invoice_line = invoice_lines.browse(row['base_line_id'])
+                tax_line = invoice_lines.browse(row['tax_line_id'])
+                src_line = invoice_lines.browse(row['src_line_id'])
+
+                tax = self.env['account.tax'].browse(row['tax_id'])
+
+                invoice_lines_tax_values_dict[invoice_line].append({
+                    'base_line_id': invoice_line,
+                    'tax_line_id': tax_line,
+                    'src_line_id': src_line,
+                    'tax_id': tax,
+                    'src_tax_id': self.env['account.tax'].browse(row['group_tax_id']) if row['group_tax_id'] else tax,
```

Comment:
Why don't you do the same as for `invoice_line`, `tax_line`, `src_line` and `tax` ?

Reply:
That makes sense - I don't know why it was this way (I just reused the existing code), but I've changed it now.